siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+        with streamlit.form("nodes"):
+            option = streamlit.selectbox('Pick a node to inspect',
+                                         metric_dataframe.columns.tolist())
+
+            params_submitted = streamlit.form_submit_button("Run")
+            if not params_submitted and node_from_flowgraph is not None:
+                option = node_from_flowgraph
+                streamlit.session_state['selected'] = []
+            if params_submitted:
+                streamlit.session_state['selected'] = []
+    return option
+
+
+def show_dataframe_and_parameter_selection(metric_dataframe):
+    """
+    Displays multi-select check box to the users which allows them to select
+    which nodes and metrics to view in the dataframe
+
+    Args:
+        metric_dataframe (Pandas.DataFrame) : contains the metrics of all nodes
+    """
+    container = streamlit.container()
+
+    transpose = streamlit.session_state['transpose']
+
+    if transpose:
+        # put data back to normal if need be
+        metric_dataframe = metric_dataframe.transpose()
+
+    displayToData = {}
+    displayOptions = []
+
+    if transpose:
+        for metric, unit in metric_dataframe.columns.tolist():
+            displayToData[metric] = (metric, unit)
+            displayOptions.append(metric)
+    else:
+        for metric, unit in metric_dataframe.index.tolist():
+            displayToData[metric] = (metric, unit)
+            displayOptions.append(metric)
+
+    options = {'cols': metric_dataframe.columns.tolist(),
+               'rows': metric_dataframe.index.tolist()}
+
+    # pick parameters
+    with streamlit.expander("Select Parameters"):
+        with streamlit.form("params"):
+            if transpose:
+                nodes = streamlit.multiselect('Pick nodes to include',
+                                              metric_dataframe.index.tolist(),
+                                              [])
+                options['rows'] = nodes
+
+                metrics = streamlit.multiselect('Pick metrics to include?',
+                                                displayOptions,
+                                                [])
+                options['cols'] = []
+                for metric in metrics:
+                    options['cols'].append(displayToData[metric])
+            else:
+                display_dataframe = metric_dataframe.columns.tolist()
+                nodes = streamlit.multiselect('Pick nodes to include',
+                                              display_dataframe,
+                                              [])
+                options['cols'] = nodes
+
+                metrics = streamlit.multiselect('Which metrics to include?',
+                                                displayOptions,
+                                                [])
+                options['rows'] = []
+                for metric in metrics:
+                    options['rows'].append(displayToData[metric])
```

Comment:
seems the only difference is options['rows/cols'] so maybe this canbe simplified to:
```
if transpose:
  key0 = 'rows'
  key1 = 'cols'
else:
  key0 = 'cols'
....
```